crypto/src/ed25519.rs

```diff
@@ -264,14 +278,73 @@ impl<'de> Deserialize<'de> for Ed25519Signature {
     where
         D: serde::Deserializer<'de>,
     {
-        let bytes = if deserializer.is_human_readable() {
-            let s = String::deserialize(deserializer)?;
-            base64ct::Base64::decode_vec(&s).map_err(|e| de::Error::custom(e.to_string()))?
+        struct Ed25519SignatureVisitor {
```

Comment:
If we just put `#[derive(Deserialize, Serialize)]` on top of `Ed25519Signature` and `#[serde(skip)]` on top of `bytes`, we can save the custom ser/de logic here. The downsides are:
1. Human readable data is not base64, but something less compact.
2. We will not be using the `bytes` once_cell "cache".

Reply:
> This is excellent!
> 
> I think that the only thing that is missing is a few lines of english in a comment of `impl (De)Serialize` summarizing what invariants the programmer should maintain when editing the inner representation of our types:
> 
> * we use this disjunction between human-readable and non-human readable formats since #460,
> * at the same time we aim to have serde_reflection processable types, which works poorly [with aliases and nameless types](https://docs.rs/serde-reflection/latest/serde_reflection/index.html#unsupported-idioms).
> 
> Here's why I think a comment about this is necessary: you're making specific choices here (e.g. the marker fields for disjunction between b64 and raw formats) that we may debate in the future. If someone was to make different ones, what shouldn't they break?

Thanks for the feedback. Added a comment with the information outlined above.